core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -137,6 +153,73 @@ class AdminClient(val time: Time,
       broker -> Try[NodeApiVersions](new NodeApiVersions(getApiVersions(broker).asJava))
     }.toMap
 
+
+  /*
+   * Remove all the messages whose offset is smaller than the given offset of the corresponding partition
+   *
+   * PurgeDataResult contains either lowWatermark of the partition or exception. We list the possible exception
+   * and their interpretations below:
+   *
+   * - DisconnectException if leader node of the partition is not available. Need retry by user.
```

Comment:
Discussed offline. Given that existing APIs such as `AdminClient.listGroupOffsets(..)` and `KafkaConsumer.retrieveOffsetsByTimes(..)` would not retry and just expose the exception to user, it should be OK for `AdminClient.purgeDataBefore(...)` to do the same thing. If we think it is useful to retry for user, we probably want to discuss the new configs and do it in Java AdminClient as part of KIP-117.